File: samples/sandbox/app/src/main/java/com/scichart/docsandbox/examples/javaBuilder/advanced2dTopics/ViewportManager.java
package com.scichart.docsandbox.examples.javaBuilder.advanced2dTopics;

import androidx.annotation.NonNull;

import com.scichart.charting.viewportManagers.DefaultViewportManager;
import com.scichart.charting.visuals.SciChartSurface;
import com.scichart.charting.visuals.axes.IAxisCore;
import com.scichart.data.model.IRange;
import com.scichart.docsandbox.core.ExampleDefinition;
import com.scichart.docsandbox.examples.base.SingleChart2DFragment;

@ExampleDefinition()
public class ViewportManager extends SingleChart2DFragment {
    @Override
    protected void initExample(@NonNull SciChartSurface surface) { }

    void setViewportManager(@NonNull SciChartSurface surface) {
        // <SetViewportManager>
        // Assume a surface has been created and configured somewhere
        surface.setViewportManager(new DefaultViewportManager());
        // </SetViewportManager>
    }

    void createCustomViewportManager() {
        // <CreateCustomViewportManager>
        class CustomViewportManager extends DefaultViewportManager {
            @Override
            protected void onUpdateXAxis(IAxisCore xAxis) {
                super.onUpdateXAxis(xAxis);
                // called before drawing of xAxis
                // here you can update visible range
                final IRange visibleRange = xAxis.getVisibleRange();
            }

            @Override
            protected void onUpdateYAxis(IAxisCore yAxis) {
                super.onUpdateYAxis(yAxis);
                // called before drawing of yAxis
                // here you can update visible range
                final IRange visibleRange = yAxis.getVisibleRange();
            }
            @Override
            protected void onApplyAutoRange(IAxisCore axis) {
                super.onApplyAutoRange(axis);
                // called when axis uses AutoRange.Always or AutoRange.Once
                // here you can update visible range when need to perform auto range
                final IRange visibleRange = axis.getVisibleRange();
            }
        }
        // </CreateCustomViewportManager>
    }
}

File: samples/sandbox/app/src/main/java/com/scichart/docsandbox/examples/kotlin/advanced2dTopics/ViewportManager.kt
package com.scichart.docsandbox.examples.kotlin.advanced2dTopics

import com.scichart.charting.viewportManagers.DefaultViewportManager
import com.scichart.charting.visuals.SciChartSurface
import com.scichart.charting.visuals.axes.IAxisCore
import com.scichart.docsandbox.core.ExampleDefinition
import com.scichart.docsandbox.examples.base.SingleChart2DFragment

@ExampleDefinition
class ViewportManager : SingleChart2DFragment() {
    override fun initExample(surface: SciChartSurface) {}

    fun setViewportManager(surface: SciChartSurface) {
        // <SetViewportManager>
        // Assume a surface has been created and configured somewhere
        surface.viewportManager = DefaultViewportManager()
        // </SetViewportManager>
    }

    fun createCustomViewportManager() {
        // <CreateCustomViewportManager>
        class CustomViewportManager : DefaultViewportManager() {
            override fun onUpdateXAxis(xAxis: IAxisCore) {
                super.onUpdateXAxis(xAxis)
                // called before drawing of xAxis
                // here you can update visible range
                val visibleRange = xAxis.visibleRange
            }

            override fun onUpdateYAxis(yAxis: IAxisCore) {
                super.onUpdateYAxis(yAxis)
                // called before drawing of yAxis
                // here you can update visible range
                val visibleRange = yAxis.visibleRange
            }

            override fun onApplyAutoRange(axis: IAxisCore) {
                super.onApplyAutoRange(axis)
                // called when axis uses AutoRange.Always or AutoRange.Once
                // here you can update visible range when need to perform auto range
                val visibleRange = axis.visibleRange
            }
        }
        // </CreateCustomViewportManager>
    }
}
